### PR TITLE
Make deploy.sh and delete.sh run properly on MacOS

### LINKED ISSUE #152: Make deploy.sh and delete.sh run properly on MacOS
I notice that `getopt` is used in both `deploy.sh` and `delete.sh`, but macOS does not support this command by default.
Can we use another solution to replace `getopt`? Such as this:

```bash
while test $# -gt 0
do
  _key="$1"
  case "$_key" in
    --all)
      all=true
      shift
      ;;
    --with-shipwright)
      with_shipwright=true
      shift
      ;;
    --with-knative)
      with_knative=true
      shift
      ;;
    --with-openFuncAsync)
      with_openFuncAsync=true
      shift
      ;;
    -p | --poor-network)
      poor_network=true
      shift
      ;;
    *)
      echo "Internal error!"
      exit 1
      ;;
  esac
  shift
done
```

### PR DESCRIPTION
**Pull request by arugal**
Signed-off-by: zhangwei <zhangwei24@apache.org>

fix #152

### CODE REVIEW

**Comment by arugal** [comment]
It appears that the [github action](https://github.com/OpenFunction/OpenFunction/blob/76692ca583343d182dd67868919901ed417d7fc6/.github/workflows/main.yaml#L23-L31) does not include the hack path.

**Comment by wanjunlei** [comment]
Yes, I will fix it later.

File: hack/delete.sh [hunk 1]
```diff
@@ -55,6 +48,12 @@ if [ "$all" = "true" ]; then
   with_openFuncAsync=true
 fi
 
+if [ "$poor_network" = "false" ]; then
```
Comment (wanjunlei) [inline hack/delete.sh:51]: Currently, the installation of `cert-manager` is placed in `bundle.yaml`, so the delete script should not delete `cert-manager`

Comment (arugal) [reply hack/delete.sh]: done

**Comment by wanjunlei** [comment]
The github action run failed, please take a look. It seems like only one parameter can be recognized for the script.

**Comment by arugal** [comment]
> The github action run failed, please take a look. It seems like only one parameter can be recognized for the script.

Done, remove some extra `shift` :)

**Review by wanjunlei — APPROVED** [review]

**Pull request merged**